db/migrations/20210129201213_single_name_field.sql

```diff
@@ -0,0 +1,15 @@
+-- migrate:up
+-- Add new single name field
+ALTER TABLE users ADD COLUMN name VARCHAR;
```

Comment:
Do we also want to offer `first_name` and `last_name` generated columns? This may be wrong sometimes, but would be quite handy a lot of the time. Adding some SQL `COMMENT`s about how they're wrong would help.

```sql
first_name GENERATED ALWAYS AS (COALESCE(preferred_name, SPLIT_PART(full_name, ' ', 1))) VIRTUAL;
last_name GENERATED ALWAYS AS (SPLIT_PART(full_name, ' ', 2)) VIRTUAL;
```

This is a slightly modified version of the SQL I sent you the other day, with the addition that `first_name` will choose the `preferred_name` if it's not null.

Reply:
Evidently Postgres doesn't support virtual columns yet, but they do support generated columns which is effectively the same thing. https://www.postgresql.org/docs/12/ddl-generated-columns.html

Reply:
Huh, a weird omission. Not important though, just replace `VIRTUAL` with `STORED` in that snippet and it'll function the same (from the query's perspective).

Reply:
> Won't `SPLIT_PART(full_name, ' ', 2)` just give us the second token in the `full_name` field? I'm going to try to split the name, take all the elements of the array except the first, and join that into a string.

Yeah, I think you're right about that.

Reply:
Yeah, I got this working but I'm not sure it is worth it.

```sql
ALTER TABLE users
  ADD COLUMN last_name VARCHAR
    GENERATED ALWAYS AS (array_to_string((regexp_split_to_array(full_name, E'\\s+'))[2:], ',', '*')) STORED;
```

Reply:
I'll create `display_name`  which is generated as either `preferred_name` or `full_name`.

Reply:
I thought `preferred_name` was removed?